dataprofiler/tests/profilers/profiler_options/test_unstructured_text_profile_options.py

```diff
@@ -0,0 +1,138 @@
+                          "attribute 'is_enabled'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option._set_helper({'is_case_sensitive.is_enabled': True}, 'test')
+
+        expected_error = ("type object 'test.stop_words' has no "
+                          "attribute 'is_enabled'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option._set_helper({'stop_words.is_enabled': True}, 'test')
+
+    def test_set(self):
+        option = self.get_options()
+
+        params_to_check = [
+            dict(prop='is_enabled', value_list=[False, True]),
+            dict(prop='is_case_sensitive', value_list=[False, True]),
+            dict(prop='stop_words',
+                 value_list=[None, ['word1', 'word2'], []]),
+            dict(prop='words', value_list=[False, True]),
+            dict(prop='vocab', value_list=[False, True]),
+        ]
+
+        # this code can be abstracted to limit code everywhere else
+        # AKA, params_to_check would be the only needed code plus raise errors
+        def _assert_set_helper(prop, value):
+            if prop not in ['words', 'vocab']:
+                option.set({prop: value})
+                self.assertEqual(value, getattr(option, prop), msg=prop)
+            else:
+                prop_enable = '{}.is_enabled'.format(prop)
+                option.set({prop_enable: value})
+                self.assertEqual(value, option.properties[prop].is_enabled,
+                                 msg=prop)
+
+        for params in params_to_check:
+            prop, value_list = params['prop'], params['value_list']
+            for value in value_list:
+                _assert_set_helper(prop, value)
+
+        # Treat is_case_sensitive and stop_words as BooleanOption
+        expected_error = ("type object 'is_case_sensitive' has no attribute "
+                          "'is_enabled'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'is_case_sensitive.is_enabled': True})
+
+        expected_error = ("type object 'stop_words' has no attribute "
+                          "'is_enabled'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'stop_words.is_enabled': True})
+
+    def test_validate_helper(self):
+        super(TestTextProfilerOptions, self).test_validate_helper()
+
+    def test_validate(self):
+
+        super(TestTextProfilerOptions, self).test_validate()
+
+        params_to_check = [
+            # non errors
+            dict(prop='is_enabled', value_list=[False, True], errors=[]),
+            dict(prop='is_case_sensitive', value_list=[False, True], errors=[]),
+            dict(prop='stop_words',
+                 value_list=[None, ['word1', 'word2'], []],
+                 errors=[]),
+            dict(prop='words', value_list=[False, True], errors=[]),
+            dict(prop='vocab', value_list=[False, True], errors=[]),
```

Comment:
shouldn't this error because words/vcab needs to be a `BooleanOption`?

Reply:
e.g.
```python
dict(prop='words', value_list=[BooleanOption(is_enabled=False), 
                               BooleanOption(is_enabled=True)], errors=[]),
dict(prop='vocab', value_list=[BooleanOption(is_enabled=False), 
                               BooleanOption(is_enabled=True)], errors=[]),
```

if this is done, the code below could be agnostic to parameters again and not need to be altered from its original form

Reply:
revert the function